### DISCUSSION TITLE
How to modify / debug  / develop Deployer package locally

### DISCUSSION THREAD

**Discussion by dvdinbox** [deployphp/deployer#3609] [Help needed]
I'm having an issue with this package (link below)

I'd like to try to debug and test changes to solve my issue but I can't figure out how to do that. Please note I'm relatively new to the php / composer ecosystem.

Here's what I tried.

Copied the deployer package to an alternate location and update composer.json to the following:

```
    "require-dev": {
        "deployer/deployer": "@dev",
    },
    "repositories": [
        {
            "type": "path",
            "url": "./deployer"
        }
    ],
```

updated the problem 'provision:website' task in the copied package located at ./deployer/recipe/provision/website.php with

````
task('provision:website', function () {
    info("DEBUG");
})->limit(1);
````

I delete the composer lock file and run composer install which reports

Package operations: 0 installs, 1 update, 0 removals
  - Removing deployer/deployer (v7.3.1)
  - Installing deployer/deployer (dev-deployer): Symlinking from ./deployer

----

When I run `dep provision:web` , my `    info("DEBUG");` code is not execute at all. The original issue persists

How can I debug and test changes to this package?

Related Issue:

https://github.com/deployphp/deployer/issues/3542

**Comment by antonmedv** [comment]
Deployer ships as a phar package. 

One simple way is to just redefine needed task in deploy.php. 

Another just clone the repo, composer install and use ~/path/to/deployer/bin/dep.
